src/Illuminate/Database/Eloquent/Collection.php

```diff
@@ -139,6 +139,27 @@ public function map(callable $callback)
     }
 
     /**
+     * Reload a fresh model instance from the database for all the entities.
+     *
+     * @param  array|string  $with
+     * @return static
+     */
+    public function fresh($with = [])
+    {
+        $model = $this->first();
```

Comment:
Oh yeah! I understand now. 
But why would you want to `fresh` an empty collection ? It's about loading all models from database, if you have no model, no need to fresh it ?

But anyway you're right, I will fix it by a check with a isEmpty.

Thank's!